glean-core/src/upload/mod.rs

```diff
@@ -216,6 +219,44 @@ pub struct PingUploadManager {
     in_flight: RwLock<HashMap<String, (TimerId, TimerId)>>,
 }
 
+impl MallocSizeOf for PingUploadManager {
+    fn size_of(&self, ops: &mut malloc_size_of::MallocSizeOfOps) -> usize {
+        let shallow_size = {
+            let queue = self.queue.read().unwrap();
+            if ops.has_malloc_enclosing_size_of() {
+                if let Some(front) = queue.front() {
+                    // The front element is an interior pointer.
+                    unsafe { ops.malloc_enclosing_size_of(front) }
+                } else {
+                    // This assumes that no memory is allocated when the VecDeque is empty.
+                    0
+                }
+            } else {
+                // An estimate.
+                queue.capacity() * mem::size_of::<PingRequest>()
+            }
+        };
+
+        let mut n = shallow_size
+            + self.directory_manager.size_of(ops)
+            // We own this arc
+            + unsafe { ops.malloc_size_of(self.processed_pending_pings.as_ptr()) }
```

Comment:
If this is the safety rationale, we still need it to conform to the format of other safety notes.